#441 part 1 application page LOQ

andreievg opened the pull request:
branched from: #314-duplicate-application-responses
requires back end: #314fe-requires-restart-status

the purpose of this branch was to give an introduction (via the Testing branch) to how work up to this branch can be used to full-fill #441 (application page and application summary for LOQ) and #437 (application home for change requested)

### Changes

* slight change to progress bar, something I've noticed when testing, when anything on a page is not valid we should show it as such (rather then blue circle)

### Testing

use branch #441-TESTING-part-1-application-page-LOQ
and back end branch above

create a review for "ABC123" with LOQ

the go to application home 

* notice that header is showing that 'there are issues'
* notice button you can click to start change requested (this should only be clicked once, after that use resume button)
* press CheckChangesRequested
* Notice progress bar statuses and json under the elements
* play around with changing things and trying to submit (logic should work just like it did previously in non linear, will consider change request that are not done as invalid and incomplete)


Here is the diff for reference: `https://github.com/openmsupply/application-manager-web-app/compare/%23441-part-1-application-page-LOQ...%23441-TESTING-part-1-application-page-LOQ`

nmadruga commented:
Wow! Tested this one briefly and it looks great.

I'll have a look on the code now, I think the 3 PRs are very dependent, but ideally we need the fix for at least the last one I noticed and added this [comment]() to stop the loop when loading the page with Changes requested so we can add one by one. Or if you think it's quicker to just make changes to this PR and we can do any more refinements only on this one and merge the others as is, will be good for me too - I think the later solution will make this easier to follow.

nmadruga reviewed (approved):
Looks good to me. I think once you have addressed comments done on #445 these is good to go. I'll have a quick look on the one before this as well.

andreievg commented:
> Wow! Tested this one briefly and it looks great.

thanks

> I'll have a look on the code now, I think the 3 PRs are very dependent, but ideally we need the fix for at least the last one I noticed and added this comment to stop the loop when loading the page with Changes requested so we can add one by one. Or if you think it's quicker to just make changes to this PR and we can do any more refinements only on this one and merge the others as is, will be good for me too - I think the later solution will make this easier to follow.

nice find, i fixed it there
https://github.com/openmsupply/application-manager-web-app/pull/445#issuecomment-800823866
